src/systems/acoustic_comms/AcousticComms.cc

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * Development of this module has been funded by the Monterey Bay Aquarium
+ * Research Institute (MBARI) and the David and Lucile Packard Foundation
+ */
+#include <unordered_map>
+
+#include <gz/common/Profiler.hh>
+#include <gz/plugin/Register.hh>
+#include <gz/sim/Entity.hh>
+#include "gz/sim/comms/Broker.hh"
+#include "gz/sim/comms/MsgManager.hh"
+#include "gz/sim/Util.hh"
+#include "AcousticComms.hh"
+
+using namespace gz;
+using namespace sim;
+using namespace systems;
+
+class AcousticComms::Implementation
+{
+  // Default max range for acoustic comms in metres.
+  public: double maxRange = 1000.0;
+
+  // Default speed of sound in air in metres/sec.
+  public: double speedOfSound = 343.0;
+
+  // Position of the transmitter at the time the message was
+  // sent, or first processed.
+  public: std::unordered_map
+          <std::shared_ptr<msgs::Dataframe>, math::Vector3d>
+          poseSrcAtMsgTimestamp;
+};
+
+AcousticComms::AcousticComms()
```

Comment:
Nit: Use the delimiters on top of each function.
```
//////////////////////////////////////////////////
```

Reply:
Done : https://github.com/gazebosim/gz-sim/pull/1608/commits/3db98f2e47a03311b01cdcac79f9be6e8b14d8e3